libs/net/core/Http/HttpRequestClient.cs

```diff
@@ -78,20 +79,29 @@ public void Dispose()
             var responseStream = await response.Content.ReadAsStreamAsync();
             // var data = await response.Content.ReadAsByteArrayAsync();
             var contentType = response.Content.Headers.ContentType;
+            var body = "";
             try
             {
-                if (contentType?.MediaType?.Contains("json", StringComparison.InvariantCultureIgnoreCase) == true)
+                if (contentType?.MediaType?.Contains("json", StringComparison.InvariantCultureIgnoreCase) == true && response.IsSuccessStatusCode)
                     return JsonSerializer.Deserialize<TModel>(responseStream, _serializeOptions);
+                else if (response.IsSuccessStatusCode)
+                    return default;
+                else if (response.StatusCode == HttpStatusCode.NotModified)
+                    return default;
             }
             catch (Exception ex)
             {
+                // Catch deserialization errors and log the response data.
                 var data = responseStream.ReadAllBytes();
-                var body = Encoding.Default.GetString(data);
+                body = Encoding.Default.GetString(data);
                 _logger.LogError(ex, "Failed to deserialize response: {body}", body);
                 throw;
             }
 
-            throw new HttpClientRequestException(response, $"Response must contain JSON but was '{contentType?.MediaType}'.");
```

Comment:
This was causing an exception due to attempting to read the body of the response twice.